src/ast/passes/codegen_llvm.cpp

```diff
@@ -3054,81 +3061,99 @@ Value *CodegenLLVM::createLogicalOr(Binop &binop)
 Function *CodegenLLVM::createLog2Function()
 {
   auto ip = b_.saveIP();
-  // log2() returns a bucket index for the given value. Index 0 is for
-  // values less than 0, index 1 is for 0, and indexes 2 onwards is the
-  // power-of-2 histogram index.
+  // Arguments: N (int64), K (0..5)
+  // Maps each power of 2 into 2^K buckets, so we can build fine-grained
+  // histograms with low runtime cost.
+  //
+  // Returns 0 for N < 0, 1..2^K for 0 <= N < 2^K; otherwise
+  // N has the form ..01xxx... where the leftmost is at bit L >= K,
+  // xxx are the following K bits with a value 0 <= X < 2^K.
+  // In this case the function returns 1 + (L-K + 1) * 2^K + X
```

Comment:
Similar to comment on `output.cpp`, I'm finding this hard to understand. The `..` in `..01` is confusing to me - is this supposed to be two bits or an arbitrary number of bits?

And "leftmost" and "L" are similarly ambiguous. Leftmost what? 

Also the "In this case" bit is confusing. What other cases are there?

Basically what I'm saying is that a very verbose explanation would be really helpful for mere mortals like me (who would otherwise have no hope of ever understanding given any reasonable amount of time)

Reply:
I added a description and an example, hopefully more clear now, PTAL

Reply:
Thanks! The description and example is very helpful